File: UnitTest.cpp
#include <iostream>
#include "Solution.hpp"

struct testcase {
    vector<int> nums;
    int output;
};

class UnitTest {
private:
    vector<testcase> testcases;
    Solution obj;
public:
    UnitTest() {
        testcases = {{{10,4,-8,7}, 2}, 
                     {{2,3,1,0}, 2}};
    }

    void test() {
        for(int i = 0; i < testcases.size(); ++i) {
            int result = obj.waysToSplitArray(testcases[i].nums);
            cout << "TestCase " << i+1 << ": " << ((result == testcases[i].output)? "passed":"failed") << endl;
        }
    }
};

int main() {
    UnitTest test;
    test.test();
}

File: UnitTest.py
from timeout_decorator import timeout
from Solution import Solution
import unittest

class UnitTest(unittest.TestCase):
    def setUp(self):
        self.__testcases = {1: ([10,4,-8,7], 2), 2: ([2,3,1,0], 2)}
        self.__obj = Solution()
        return super().setUp()
    
    @timeout(0.5)
    def test_case_1(self):
        nums, output = self.__testcases[1]
        result = self.__obj.waysToSplitArray(nums = nums)
        self.assertIsInstance(result, int)
        self.assertEqual(result, output)

    @timeout(0.5)
    def test_case_2(self):
        nums, output = self.__testcases[2]
        result = self.__obj.waysToSplitArray(nums = nums)
        self.assertIsInstance(result, int)
        self.assertEqual(result, output)

if __name__ == '__main__': unittest.main()